lib/awful/rules.lua

```diff
@@ -342,10 +342,41 @@ end
 
 rules.add_rule_source("awful.rules", apply_awful_rules, {"awful.spawn"}, {})
 
+local blacklisted_snid = {}
+
+--- Make startup notification work for some clients like XTerm. This is ugly
+-- but works often enough to be useful.
```

Comment:
The commit message also talks about KF5, which seems to be part of KDE. How about fixing this in KDE / Qt, too? (Or is it at least reported as a bug somewhere?)

Reply:
Wont happen. The X11 support is frozen, the maintainer isn't interested in fixing this bug or messing with that code. I did talk to him directly about this (we also worked for the same employer for a while).

It's also not a simple fix. The Qt (X11Extra module) has support for this, but is not enabled by default. Then the KDE "QPA" (Qt Platform Plugin) for Linux also has code for this but it's under a dozen layer of abstraction and is never called due to a bunch of things that are unlikely to change.